workflow-manager/limiter/limiter.go

```diff
@@ -0,0 +1,56 @@
+package limiter
+
+import "sync"
+
+// Limiter is similar to sync.Pool, however, it limits how many shared
+// objects (in this case workers) there can be
+type Limiter struct {
+	workers chan int32
+}
+
+// Ticket is the structure that allows you to schedule jobs
+type Ticket struct {
+	worker int32
+	lock   sync.Mutex
+	dirty  bool
+}
+
+// New creates a new instance of Limiter that limits how many functions can get executed without the other ones completing first
+func New(maxWorkers int32) *Limiter {
+	// Channels are blocking in go, if you write to them, something needs to read from them
```

Comment:
This is a [comma splice](https://www.grammarly.com/blog/comma-splice/).

```suggestion
	// Channels are blocking in go. If you write to them, something needs to read from them.
```